File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/AnswerReveal.java
package tech.zolhungaj.amqapi.servercommands.gameroom.game;

import tech.zolhungaj.amqapi.servercommands.CommandType;
import tech.zolhungaj.amqapi.servercommands.objects.PlayerAnswer;
import tech.zolhungaj.amqapi.servercommands.objects.ProgressBarState;

import java.util.List;
import java.util.Optional;

@CommandType("player answers")
public record AnswerReveal(
    List<PlayerAnswer> answers,
    Optional<ProgressBarState> progressBarState
){}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/game/AnswerReveal.kt
package tech.zolhungaj.amqapi.servercommands.gameroom.game

import tech.zolhungaj.amqapi.servercommands.CommandType
import tech.zolhungaj.amqapi.servercommands.objects.PlayerAnswer
import tech.zolhungaj.amqapi.servercommands.objects.ProgressBarState

@JvmRecord
@CommandType("player answers")
data class AnswerReveal(
    val answers: List<PlayerAnswer>,
    val progressBarState: ProgressBarState?
)
